src/grimp/adaptors/importscanner.py

```diff
@@ -32,47 +46,61 @@ def scan_for_imports(
         """
         found_package = self._found_package_for_module(module)
         module_filename = self._determine_module_filename(module, found_package)
-        is_package = self._module_is_package(module_filename)
         module_contents = self._read_module_contents(module_filename)
-        module_lines = module_contents.splitlines()
 
         try:
-            ast_tree = ast.parse(module_contents)
+            imported_objects = self._get_raw_imported_objects(module_contents)
         except SyntaxError as e:
             raise exceptions.SourceSyntaxError(
                 filename=module_filename,
                 lineno=e.lineno,
                 text=e.text,
             )
```

Comment:
@seddonym  This is the part that https://github.com/Peter554/pyimportparse could replace.